woocommerce/woocommerce-android
# Feature/check iap eligibility

Part of: #7925


### Description

First step of IAP integration in store creation flow: check if user is eligible to use IAP. 

- If user is eligible, the native flow will be triggered 
- If user is not eligible, either because of an error while checking eligibility or because IAP is not supported for the Google account on the device a dialog will be shown explaining store creation flow is not available in their country. 

The feature is currently hidden under a feature flag and won't be enabled unless we are rejected from one of the app stores.  

### Testing instructions

Follow these instructions: PdfdoF-1Ej-p2 (Android section) to get an eligible account for IAP and test the feature. 
- Check that triggering store creation with an eligible account in `debug` app navigates to native store creation flow
- Check that triggering store creation in `debug` app with a non IAP eligible account displays the error dialog
- Check that if feature flag `IAP_FOR_STORE_CREATION -> false` then the native store creation flow is opened as usual

### Images/gif

Success 


https://user-images.githubusercontent.com/2663464/208727825-a08ed355-1860-4ec6-8f87-ebe5150697ac.mp4


Error

https://user-images.githubusercontent.com/2663464/208727058-feff864a-4ed8-45e3-9749-11a64a18b08e.mp4
